sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from "../util/logging";
+
+import * as child_process from "child_process";
+
+/**
+ * Provides the user access token and expire time
+ * with Azure CLI command "az account get-access-token".
+ */
+export class AzureCliCredential implements TokenCredential {
+  /**
+   * Creates an instance of the AzureCliCredential class.
+   */
+  constructor() {}
+
+  /**
+   * Gets the access token from Azure CLI
+   * @param resource The resource to use when getting the token
+   */
+  protected async getAzureCliAccessToken(resource: string) {
+    return new Promise((resolve, reject) => {
+      try {
+        child_process.exec(
```

Comment:
To address security concerns, in .NET and [Python](https://github.com/Azure/azure-sdk-for-python/pull/10092) we use a working directory on the path.

Reply:
I missed this comment, but I can do a follow-up PR